Source/WebCore/rendering/RenderBox.cpp

```diff
@@ -1950,6 +1950,16 @@ void RenderBox::paintClippingMask(PaintInfo& paintInfo, const LayoutPoint& paint
     if (!paintInfo.shouldPaintWithinRoot(*this) || style().visibility() != Visibility::Visible || paintInfo.phase != PaintPhase::ClippingMask || paintInfo.context().paintingDisabled())
         return;
 
+#if ENABLE(LAYER_BASED_SVG_ENGINE)
+    if (document().settings().layerBasedSVGEngineEnabled() && style().clipPath() && style().clipPath()->type() == PathOperation::Reference) {
+        ASSERT(hasLayer());
+        LayoutSize offsetFromRoot = layer()->offsetFromAncestor(layer()->root());
```

Comment:
It's unusual to see layer tree information being used in RenderBox to drive clipping.
I'd investigate to avoid using the layer tree, to extract the information from the box model data - paintOffset, size(), location(), ...

Reply:
Yes, it was old code, things work when using paintOffset and size() :)